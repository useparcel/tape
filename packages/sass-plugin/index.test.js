import Tape from "../tape/index.js";
import SassPlugin from "./index.js";

describe("SassPlugin", () => {
  test("should convert an external sass file to css", async () => {
    const tape = new Tape({
      entry: "/index.html",
      plugins: [SassPlugin],
      files: {
        "/index.html": {
          content: `
            <html>
              <head>
                <link rel="stylesheet" href="/style.scss">
              </head>
              <body>
                body
              </body>
            </html>
          `,
        },
        "/style.scss": {
          content: `
            $my-color: blue;

            body {
              background: $my-color;
            }
          `,
        },
      },
    });

    const results = await tape.build();
    expect(results).toMatchSnapshot();
  });

  test("should convert embedded sass to css", async () => {
    const tape = new Tape({
      entry: "/index.html",
      plugins: [SassPlugin],
      files: {
        "/index.html": {
          content: `
            <html>
              <head>
                <style type="text/scss">
                  $my-color: blue;

                  body {
                    background: $my-color;
                  }
                </style>
              </head>
              <body>
                body
              </body>
            </html>
          `,
        },
      },
    });

    const results = await tape.build();
    expect(results).toMatchSnapshot();
  });
});
