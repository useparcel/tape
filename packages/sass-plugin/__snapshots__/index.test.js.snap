// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SassPlugin should convert an external sass file to css 1`] = `
Object {
  "entry": "/index.html",
  "files": Object {
    "/index.html": Object {
      "content": "
            <html>
              <head>
                <link rel=\\"stylesheet\\" href=\\"/style.css\\">
              </head>
              <body>
                body
              </body>
            </html>
          ",
    },
    "/style.css": Object {
      "content": "body {
  background: blue;
}",
    },
  },
}
`;

exports[`SassPlugin should convert embedded sass to css 1`] = `
Object {
  "entry": "/index.html",
  "files": Object {
    "/index.html": Object {
      "content": "
            <html>
              <head>
                <style type=\\"text/scss\\">body {
  background: blue;
}</style>
              </head>
              <body>
                body
              </body>
            </html>
          ",
    },
  },
}
`;
